mqtt-app-starters-common/src/main/java/org/springframework/cloud/stream/app/mqtt/MqttProperties.java

```diff
@@ -0,0 +1,132 @@
+/*
+ * Copyright 2017 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.springframework.cloud.stream.app.mqtt;
+
+import org.springframework.boot.context.properties.ConfigurationProperties;
+
+/**
+ * Generic mqtt connection properties.
+ *
+ * @author Janne Valkealahti
+ *
+ */
+@ConfigurationProperties("mqtt")
+public class MqttProperties {
+
+	/**
+	 * location of the mqtt broker(s) (comma-delimited list)
+	 */
+	private String[] url = new String[] { "tcp://localhost:1883" };
```

Comment:
Some of the properties might be required - need particular validations to protect against misconfiguration